src/format/KeePass1Reader.cpp

```diff
@@ -859,13 +858,9 @@ bool KeePass1Reader::parseCustomIcons4(const QByteArray& data)
         if (static_cast<quint32>(data.size()) < (pos + iconSize)) {
             return false;
         }
-        QImage icon = QImage::fromData(data.mid(pos, iconSize));
+        QByteArray icon = data.mid(pos, iconSize);
         pos += iconSize;
 
-        if (icon.width() != 16 || icon.height() != 16) {
-            icon = icon.scaled(16, 16);
-        }
```

Comment:
@droidmonkey I'm realizing that this was in `src/format/KeePass1Reader.cpp`, but not in `src/format/KdbxXmlReader.cpp`, Do you know why? Right now I moved it to `gui/Icons`, but I realize it might not be appropriate because it will be applied on all the generated icons, whichever format they are coming from.

Reply:
This should actually be removed entirely, we scale icons to 128x128 when they are ingested. This should have been deleted a long while ago!

Reply:
fixed.